.circleci/config.yml

```diff
@@ -155,25 +139,16 @@ jobs:
       - *attach-to-workspace
       - run:
           name: Licences check
-          command: bundle exec rake ci:license_finder:run
+          command: make test-rake JOB="ci:license_finder:run"
```

Comment:
Why not running rake directly?

Reply:
well, `test-rake` target is kind of "run rake and if it's not installed yet, please install it as well" (it has the bundler dependencies installation as a target it depends on). 

(and also handles the proxy stuff, if env var is set)

Reply:
That I think is entirely unnecessary/. Just compose it as two steps `make bundle`, `rake ...`.

Reply:
it's kind of a utility wrapper around rake. 

and I felt I needed some mechanism that makes a little more explicit how make - rake (should) interface with one another. 

Clearly nothing stops anyone from adding a `bundle exec rake` somewhere else in the Makefile, but, there's now an established pattern / way about how to do that... if you know what i mean... a way to answer the question how does make/rake work together kind of thing.

Reply:
The answer is they don't work together. 

Once you clone the repo, you only need to run `bundle install` and then run `rake`.
Make is there for cases when you don't operate with Ruby.

Reply:
well, you can still very much go for `bundle install` and `rake` directly. 

But I see no harm in wrapping those with make too, so that you can offer one simpler interface (in this case, `make` driven). I think it is useful to have one entrypoint that can help you understand how the project is structured and I see make as a kind of executable documentation around that.

Reply:
well, if rails developers already know the rails lifecycle and standards, they don't have to use make. It is unnecessary for them, so if we only keep that POV in mind, I agree with you. 

make can help all non-rails developers who happen to come across this project. 

and I have to say that I very much like that we follow this practice of using `make` in most (if not all) of our projects - you land there without knowing what you'll find underneath, but you can use the makefile as the first layer you can start peeling off in order to understand each project separately. in fact I would go so far as to say that I think there is a growing need in the polyglot world for such an (almost) technology agnostic approach to documenting the (build, at least) lifecycle of project X that's thrown at you. 

I am not sure if my current approach does everything we need for this ^^ , but that's where I would like to end up.

Reply:
Make can help non-rails developers by providing a dev container to get working environment.

From that it is the same level field and everyone keeps using rake as rails intended. Actually, rake is going away quite a bit in Rails 5 as the main bin/rails picks up the test runner etc.

As I mentioned many times. `make` should provision the environment (start a container, maybe run bundle install). For development environment the bundle install will be different than for CI environment. We have just one CI environment, so I don't see much value in abstraction of that.
When you get working environment it is just `rake` and `bin/rails` all way along. You are also not going to abstract all rails commands into `make`.
Ultimately the ideal way of running the test suite locally should be `circleci local` if you want to execute it whole. 

I'm afraid this make madness is going to be unmaintainable in a few weeks because it is way more complicated than the current state.